### PR TITLE
New wizard spell: Curse of the Hangman

### PR DESCRIPTION
**Pull request by ghost**
image
Somewhat inspired by virology hangman symptom.
Causes all mobs in view to have most of their word letters replaced by _s.
Starts with a cooldown of 50s and retaining 12 letters, upgrades reduce it to 20s-ish and retaining about 4-5 letters.
Others must shout the letters out in order to reveal them, repeated mistakes reset this.
The player with the most correct guesses shows up on the round end leaderboard, if any.
[content]

:cl:
 * rscadd: Wizards have now learned Curse of the Hangman, a spell obscuring the letters in others' spoken words. Cure this by guessing the missing letters, but don't make mistakes, or it will reset!

### CODE REVIEW

**Comment by Eneocho** [comment]
Is the eventual outcome the guy dying from suffocation if the letters are wrong?

**Comment by ghost** [comment]
> Is the eventual outcome the guy dying from suffocation if the letters are wrong?

Nah I didn't feel like making the spell anything more than annoying

**Comment by GreatBigFailure** [comment]
Don't have time to look at the code, so to be clear:
Each cast of the spell picks some letters to be removed and each one has to be shouted and heard by the person to allow them to speak properly again? So if I shouted CAT and C + T were in the list, would it count both? Or does it need to be C then T exclusively in each speech instance? 
And it's an AOE similar to fist? Not targeted like shoe snatch? What if you cast it on the same person twice?

**Comment by Eneocho** [comment]
>
>Nah I didn't feel like making the spell anything more than annoying

Perhaps it could be lethal if upgraded, say, for 30 points.

**Comment by ghost** [comment]
> Each cast of the spell picks some letters to be removed and each one has to be shouted and heard by the person to allow them to speak properly again? So if I shouted CAT and C + T were in the list, would it count both? Or does it need to be C then T exclusively in each speech instance?

Only checks if one letter is said at a time
> And it's an AOE similar to fist? Not targeted like shoe snatch? What if you cast it on the same person twice?

It just scrambles and resets it again

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -1980,15 +1980,30 @@ mob/living/carbon/human/isincrit()
...
+	var/hangman_score = 0 // For round end leaderboards
+
 /mob/living/carbon/human/Hear(var/datum/speech/speech, var/rendered_speech="")
 	..()
 	if(stat)
 		return //Don't bother if we're dead or unconscious
 	if(ear_deaf || speech.frequency || speech.speaker == src)
 		return //First, eliminate radio chatter, speech from us, or wearing earmuffs/deafened
+	var/mob/living/H = speech.speaker
+	if(muted_letters && muted_letters.len && length(speech.message) == 1)
```
Comment (Kurfursten) [inline code/modules/mob/living/carbon/human/human.dm]: Guessing with any punctuation (e.g.: `A!`) will count as a non-guess

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -1980,15 +1980,30 @@ mob/living/carbon/human/isincrit()
...
+		if(speech.message in muteletters_check)
+			muted_letters.Remove(speech.message)
+			H.visible_message("<span class='notice'>[speech.speaker] guessed a correct letter!</span>","<span class='notice'>You guessed a correct letter!</span>")
```
Comment (Kurfursten) [inline code/modules/mob/living/carbon/human/human.dm]: This feels like the kind of feedback I'd receive from an official hangman PDA app. It seems... weird to have this displayed from a human, presumably by magical forces. I might go with more magical-seeming span classes and make the messages themselves more enchanted sounding.

File: code/modules/mob/living/carbon/human/human.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/human.dm]: Maybe I'll try strip that from speech messages or something.

Comment (ghost) [reply code/modules/mob/living/carbon/human/human.dm]: Done

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (ghost) [reply code/modules/mob/living/carbon/human/human.dm]: Changed it

**Pull request merged**